decidim-proposals/app/services/decidim/proposals/proposal_search.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+module Decidim
+  module Proposals
+    # A service to encapsualte all the logic when searching and filtering
+    # proposals in a participatory process.
+    class ProposalSearch
```

Comment:
I agree with @mrcasals but I believe this wasn't possible in this scenario because of that weird `load` there. Still might we worth to try it.